src/main/java/com/minecolonies/coremod/colony/permissions/PermissionsView.java

```diff
@@ -0,0 +1,218 @@
+import com.minecolonies.api.util.Utils;
+import com.minecolonies.coremod.network.PacketUtils;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraftforge.fml.common.network.ByteBufUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * A client side representation of the permissions.
+ */
+public class PermissionsView implements IPermissions
+{
+    @NotNull
+    private final Map<UUID, Player>  players     = new HashMap<>();
+    @NotNull
+    private final Map<Rank, Integer> permissions = new EnumMap<>(Rank.class);
+    private       Rank               userRank    = Rank.NEUTRAL;
+
+    /**
+     * Getter for the user rank.
+     * @return the rank.
+     */
+    public Rank getUserRank()
+    {
+        return userRank;
+    }
+
+    /**
+     * Gets all player by a certain rank.
+     *
+     * @param rank the rank.
+     * @return set of players.
+     */
+    @NotNull
+    public Set<Player> getPlayersByRank(final Rank rank)
+    {
+        return Collections.unmodifiableSet(
+                this.players.values()
+                        .stream()
+                        .filter(player -> player.getRank() == rank)
+                        .collect(Collectors.toSet()));
+    }
+
+    @NotNull
+    public Map<UUID, Player> getPlayers()
+    {
+        return Collections.unmodifiableMap(players);
+    }
+
+    /**
+     * Gets all player by a certain set of rank.
+     *
+     * @param ranks the set of rank.
+     * @return set of players.
+     */
+    @NotNull
+    public Set<Player> getPlayersByRank(@NotNull final Set<Rank> ranks)
+    {
+        return Collections.unmodifiableSet(
+                this.players.values()
+                        .stream()
+                        .filter(player -> ranks.contains(player.getRank()))
+                        .collect(Collectors.toSet()));
+    }
+
+    @NotNull
+    public Map<Rank, Integer> getPermissions()
+    {
+        return permissions;
+    }
+
+    /**
+     * Checks if the player has the permission to do an action.
+     *
+     * @param id     the id of the player.
+     * @param action the action he is trying to execute.
+     * @return true if so.
+     */
+    public boolean hasPermission(final UUID id, @NotNull final Action action)
+    {
+        return hasPermission(getRank(id), action);
+    }
+
+    /**
+     * Checks if the rank has the permission to do an action.
+     *
+     * @param rank   the rank of the player.
+     * @param action the action he is trying to execute.
+     * @return true if so.
+     */
+    public boolean hasPermission(final Rank rank, @NotNull final Action action)
+    {
+        return (rank == Rank.OWNER && action != Action.GUARDS_ATTACK)
+                || (permissions != null && action != null && permissions.containsKey(rank) && Utils.testFlag(permissions.get(rank), action.getFlag()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2589)